### PR TITLE
Add support for Vault TOTP secrets

### LINKED ISSUE #7865: Vault TOTP Support
**Description**

Add support for the Vault TOTP Secret engine.

**Implementation ideas**

Create a `VaultTOTPSecretEngine` class to deal with TOTP Secret API.

@vsevel @gsmet If you agree I am happy to provide a PR about this feature. I am going to start working today a bit.

Please if you can assign me the issue :)

### PR DESCRIPTION
**Pull request by lordofthejars**
PR for adding support for Vault TOTP: https://www.vaultproject.io/api/secret/totp/index.html

When the PR is merged I'll send another PR with a new document explaining how to use it.

Fixes #7865

### CODE REVIEW

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java
Comment (sberyozkin) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java]: Can the dynamic key operations be of the practical interest to the application code ?

Comment (lordofthejars) [reply extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java]: This key is a TOTP key with different properties than a standard key, in this case, a key just contains two fields a QR code and a URL.

**Comment by sberyozkin** [comment]
@lordofthejars IMHO it is ready to go, all looks quite perfect to me, waiting for @vsevel 's feedback, cheers

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/OkHttpVaultClient.java
Comment (vsevel) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/OkHttpVaultClient.java]: I have used `body` for local variables and parameters throughout this class for conciseness. same for `result` in `VaultTOTPManager`

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/totp/VaultTOTPCreateKeyBody.java
Comment (vsevel) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/totp/VaultTOTPCreateKeyBody.java]: It is funny I have never spotted that I was sending `null` to vault in the other use cases. 
I would suggest making it a global default for the extension with `mapper.setSerializationInclusion(Include.NON_NULL);` rather than having to specify it on each field.

**Comment by lordofthejars** [comment]
@vsevel Added the comments you mentioned.

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/totp/VaultTOTPCreateKeyBody.java
Comment (vsevel) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/totp/VaultTOTPCreateKeyBody.java]: `(exported == null || exported) && (generate != null && generate)`
it is simpler, and the logic is not broken when the user passes `new Boolean(true)` instead of `Boolean.TRUE`
or (a bit longer but a bit clearer):
```
    public boolean isProducingOutput() {
        return is(exported, true) && is(generate, false);
    }

    private boolean is(Boolean v, boolean defaultValue) {
        return v == null ? defaultValue : v;
    }
```

Comment (lordofthejars) [reply extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/totp/VaultTOTPCreateKeyBody.java]: Good catch thanks.

File: integration-tests/vault/src/test/java/io/quarkus/vault/VaultTOTPITCase.java
Comment (vsevel) [inline integration-tests/vault/src/test/java/io/quarkus/vault/VaultTOTPITCase.java]: create a constant?

**Comment by vsevel** [comment]
hello I have added only minor comments. `mapper.setSerializationInclusion(Include.NON_NULL)` is the only one that could break the other use cases, but should not. let's wait for the CI.

**Comment by lordofthejars** [comment]
@vsevel Last commit with latest changes on test side, now let's wait until the build result. Locally, it worked.

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java
Comment (vsevel) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java]: actually I do not think you have any config properties.

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultTOTPManager.java
Comment (vsevel) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultTOTPManager.java]: I would rename `vaultTOTPCreateKeyBody` into `body` as well here.

**Comment by lordofthejars** [comment]
@vsevel Second round done :)

**Comment by sberyozkin** [comment]
Hi Vincent @vsevel, please approve when you are totally happy with the PR and I'll be glad to merge

**Comment by lordofthejars** [comment]
@sberyozkin @vsevel Same here please :) https://github.com/quarkusio/quarkus/pull/7891

**Review by vsevel — APPROVED** [review]

**Comment by vsevel** [comment]
@lordofthejars I just approved. very nice addition to the vault extension. thanks.

**Comment by lordofthejars** [comment]
Thank you very much to you for starting the vault extension and reviewing the PR.

**Review by gsmet — CHANGES_REQUESTED** [review]
I will have a quick look today, given Vincent and Sergey already are OK with it.

Can you squash the commits though?

**Comment by lordofthejars** [comment]
@gsmet Done.

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
@gsmet Thanks

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**
